### PR TITLE
Update validation method of protocols communication test

### PR DESCRIPTION
**Pull request by jmv74211**
|Related issue|
|---|
|#1063|

## Description

This PR changes the way of validating the agent-manager communication in test agent protocols communication. Now, instead of checking `archives.log` it is checked the `queue` socket. This change allows us to more accurately validate that the message has reached the manager, has been decoded correctly and has been written to the socket queue for `analysisd` to read later. Previously, when monitoring the archives.log we were checking one step further.

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.

- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`
- [x] Python codebase is documented following the Google Style for Python docstrings.
- [x] The test is documented in wazuh-qa/docs

### CODE REVIEW

File: docs/tests/integration/test_remoted/test_agent_communication/test_protocols_communication.md [hunk 1]
```diff
@@ -46,5 +46,18 @@ a difference of 2 seconds.
 - **TCP,UDP and port 56000**
 - **UDP,TCP and port 56000**
 
+## Comments
+
+An important aspect to take into account is the time in which remoted makes a reload of the `client.keys` info.
```
Comment (BraulioV) [inline docs/tests/integration/test_remoted/test_agent_communication/test_protocols_communication.md:51]:
```suggestion
An important aspect to take into account is the time needed by wazuh-remoted to reload the `client.keys`.
```

File: deps/wazuh_testing/wazuh_testing/remote.py [hunk 2]
```diff
@@ -338,6 +337,8 @@ def wait_to_remoted_key_update(wazuh_log_monitor):
     Raises:
         TimeoutError: if could not find the remoted key loading log.
     """
+    file.truncate_file(LOG_FILE_PATH)
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/remote.py:340]: Can you add a comment here explaining why it is important to truncate the log before waiting for the reload of the client.keys?

**Review by BraulioV — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/remote.py (hunk 2 above)
Comment (jmv74211) [reply deps/wazuh_testing/wazuh_testing/remote.py:342]: Done in https://github.com/wazuh/wazuh-qa/pull/1101/commits/fd00b88abcd5a4dcf3af786dc0a7f86fb3d3a662

File: docs/tests/integration/test_remoted/test_agent_communication/test_protocols_communication.md (hunk 1 above)
Comment (jmv74211) [reply docs/tests/integration/test_remoted/test_agent_communication/test_protocols_communication.md]: Done in https://github.com/wazuh/wazuh-qa/pull/1101/commits/fd00b88abcd5a4dcf3af786dc0a7f86fb3d3a662

**Review by BraulioV — APPROVED** [review]

**Pull request merged**